app/models/chargeback.rb

```diff
@@ -86,17 +86,20 @@ def get_rates(perf)
     @rates ||= {}
     @enterprise ||= MiqEnterprise.my_enterprise
 
+    state = perf.resource.vim_performance_state_for_ts(perf.timestamp)
```

Comment:
this is still bad performance. itll run `vim_performance_state_for_ts` once per unique timestamp+image meaning if we have metrics for one week from 4 containers with different images then `vim_performance_state_for_ts` will be called  (7days) * (24 hours) * (2 number of different images) which is still a lot.